### DISCUSSION TITLE
Disable all queries in the application

### DISCUSSION THREAD

**Discussion by AndrianStoikov** [TanStack/query#2437] [Q&A] [answered]
Hello,

I am using react-query for all API calls happening in an application. However, there are some cases that these API calls might fail, for example, the authentication of the user is no longer valid. Is it possible to catch that error and disable all queries that are in the application. Currently, I am using `queryClient.clean()` inside an `ErrorBoundary`, but I feel like the `clean` method is not what I am looking for. Is there a better way to achieve what I am trying to do? Also, is there a better way to do it?

Best regards,
Andy

**Comment by TkDodo** [comment, marked as answer]
make the authentication part of the queryKey, set it to `null` if it is no longer valid and set `enabled` to `false`  when `authentication` is `null` :)

**Reply by AndrianStoikov** [reply]
Thanks for your reply. I followed a similar approach but connected the useQuery to our generic redux store. So I could control it from there. Looks like this is the right approach.

**Reply by benhylak** [reply]
I wish there was a cleaner way to do this. Every single query I write needs to have the same boiler plate (isAuthenticated, enabled, etc.)
